File: Day 132/homework/homework.js
// 1) შექმენით ობიექტი სახელად book, რომელსაც ექნება თვისებები: title, author და year შემდეგ, console.log()-ით გამოიტანეთ წიგნის ავტორი.
const book = {
    title: "iuri boyka",
    author: "boyka",
    year:  "2016"
}

console.log(book)
console.log(book.author)

console.log("--------------------------------")

// 2) გამოიყენეთ წინა დავალებაში შექმნილი book ობიექტი. შეცვალეთ წიგნის year property უფრო ახალი წლით და დაბეჭდეთ მთლიანი ობიექტი კონსოლში.
book.year = "2017"

console.log(book)
console.log(book.year)

console.log("--------------------------------")

// 3) შექმენით ობიექტი user, რომელსაც ექნება თვისებები name და age. შემდეგ, დაამატეთ ამ ობიექტს ახალი თვისება email. ბოლოს, დაბეჭდეთ ობიექტი.
const user = {
    name: "deme",
    age: 11
}

user.email = "qweqwe@gmail.com"

console.log(user)
console.log(user.email)

console.log("--------------------------------")

// 4) შექმენით ობიექტი calculator, რომელსაც ექნება ორი property: num1 და num2. დაამატეთ ამ ობიექტს მეთოდი add, რომელიც დააბრუნებს num1-ისა და num2-ის ჯამს. გამოიძახეთ ეს მეთოდი და დაბეჭდეთ შედეგი
const calculator = {
    num1: 60,
    num2: 7,

    add: function(){
        return this.num1 + this.num2
    }
}

console.log(calculator)
console.log(calculator.add())

console.log("--------------------------------")

// 5) შექმენით ობიექტი person, რომელსაც ექნება property firstName, lastName და მეთოდი sayHello. მეთოდმა უნდა გამოიტანოს "გამარჯობა, მე ვარ [სახელი] [გვარი]".

const person = {
    firstName: "დემეტრე",
    lastName: "სუხიტაშვილი",

    sayHello: function(){
        return `მე ვარ ${this.firstName} ${this.lastName}`
    }
}

console.log(person)
console.log(person.sayHello())

console.log("--------------------------------")

File: Day 128/homework/homework.py
# ```py
# 2) შექმენით Person კლასი (name, age).
# შემდეგ შექმენით Student კლასი, რომელიც ამატებს grade-ს და super()-ით ინიციალიზაციას აკეთებს.
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

class Student(Person):
    def __init__(self, name, age, grade):
        super().__init__(name, age)
        self.grade = grade

person = Student("demetre", 11, "10 qula")

print(person.name)
print(person.age)
print(person.grade)

# 3) შექმენით Employee კლასი (name, salary).
# შემდეგ შექმენით Manager კლასი, რომელიც მშობელ კლასს დაუმატებს department კუთვნილებას და super()-ით იღებს დანარჩენ კუთვნილებებს.
class Employee:
    def __init__(self, name, salary):
        self.name = name
        self.salary = salary

class Manager(Employee):
    def __init__(self, name, salary, departament):
        super().__init__(name, salary)
        self.departament = departament

employee = Employee("deme", 50)
manager = Manager("lika", 999999999999, "mentors departament")

print(employee.name)
print(employee.salary)
print("-------------------")
print(manager.name)
print(manager.salary)
print(manager.departament)

# 4) შექმენით Shape კლასი area() მეთოდით რომელიც დააბრუნებს საწყისად 0-ს (return 0).
# შემდეგ შექმენით Rectangle კლასი (width, height), რომელიც super()-ს გამოიყენებს და მოახდენს area() მეთოდის override-ს.
class Shape:
    def area(self):
        return 0 

class Rectangle(Shape):
    def __init__(self, width, height):
        self.width = width
        self.height = height
    
    def area(self):
        super().area()
        return self.width * self.height

rect1 = Rectangle(20, 12)

print(rect1.area)

# 5) შექმენი User კლასი (username, email).
# შექმენი Admin კლასი, რომელიც ამატებს role-ს და super()-ით იძახებს მშობლის კონსტრუქტორს.
class User:
    def __init__(self, username, email):
        self.username = username
        self.email = email

class Admin(User):
    def __init__(self, username, email, role):
        super().__init__(username, email)
        self.role = role

user = User("deme", "qweqwe@gmail.com")
admin = Admin("deme", "qweqwe@gmail.com", "admin")

print(user.username)
print(user.email)
print(admin.username)
print(admin.email)
print(admin.role)

# 6) შექმენი Book კლასი (title, author).
# შემდეგ EBook კლასი (file_size), რომელიც super()-ით იძახებს მშობლის კონსტრუქტორს და ამატებს ახალ ატრიბუტს.
class Book:
    def __init__(self, title, author):
        self.title = title
        self.author = author

class Ebook(Book):
    def __init__(self, title, author, azrze_ar_var):
        super().__init__(title, author)
        self.azrze_ar_var = azrze_ar_var

book = Ebook("qweqwe", "blabla", "azrze ar var davwero aq amitom sexede terminals")

print(book.title)
print(book.author)
print(book.azrze_ar_var)
# ```